compiler/rustc_mir/src/transform/promote_consts.rs

```diff
@@ -315,74 +326,17 @@ impl<'tcx> Validator<'_, 'tcx> {
                             BorrowKind::Shallow | BorrowKind::Unique => return Err(Unpromotable),
                         }
 
-                        // We can only promote interior borrows of promotable temps (non-temps
-                        // don't get promoted anyway).
+                        // We only actually promote the projection base, so only that is further validated.
+                        // Make sure it does not need dropping!
```

Comment:
I *think* this is correct, based on the fact that we can promote this:
```rust
const ALL_THE_NUMS: [u32; 1] = [
    1
];

#[inline(never)]
fn array(i: usize) -> &'static u32 {
    return &ALL_THE_NUMS[i];
}
```
But maybe we are even more clever, and promote those projections that we can, stopping at `Index`?